java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -3442,4 +3446,58 @@ public void objectRefcount_returns_success() {
         assertEquals(testResponse, response);
         assertEquals(refcount, payload);
     }
+
+    @SneakyThrows
+    @Test
+    public void geoadd_returns_success() {
+        // setup
+        String key = "testKey";
+        Map<String, GeospatialData> membersToGeoSpatialData =
+                Map.of("Place", new GeospatialData(10.0, 20.0));
+        String[] arguments = new String[] {key, "10.0", "20.0", "Place"};
+        Long value = 1L;
+
+        CompletableFuture<Long> testResponse = new CompletableFuture<>();
+        testResponse.complete(value);
+
+        // match on protobuf request
+        when(commandManager.<Long>submitNewCommand(eq(GeoAdd), eq(arguments), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<Long> response = service.geoadd(key, membersToGeoSpatialData);
+        Long payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(value, payload);
+    }
+
```

Comment:
good to have an UT which verfies how `GeoAddOptions` converted to args
see example for `xadd` and `xtrim` options